### ISSUE TITLE
Ground speed vs. speed

### ISSUE THREAD

**Issue by martonmiklos** [jdrescher2006/Laufhelden#171]
Hello @jdrescher2006 

I notified something weird: I found some points in the training data (by scrolling the diagrams) which were showing higher speed than the maximum speed at the details page.

By peeking into the code it turned out  that maximum speed at the details page is calculated from the maximum of the ground speed, while the diagram data is populated with a speed which is calculated by the time and position difference of the points of the track. 

What was the intention behind this? I have plotted the ground speed on the diagrams and it seems to be less filtered, however sometimes the calculated speed shows really weird numbers (200 km/h with a bike is highly unlikely :laughing: )

**Comment by jdrescher2006** [comment]
HI,
i was struggling with the GPS data all the time. The filtering is not very good in Laufhelden.
I can't remember why it is done like this, I think i wasn't really satisfied with using the ground speed due to inaccuracy.

**Comment by martonmiklos** [comment]
Yeah, I plotted ground speed and it was way more furry than the speeds calculated from the position shifts. I remember for sure that in the meerun had some filtering settings for each exercise types. I might ask Benoit about this, and might look for improvements.

**Comment by jdrescher2006** [comment]
Meerun had far better filtering. I think there was a website from the developer where he explained what he did to filter the GPS data.
If you could gain some knowledge about this it would really be fine.

**Comment by martonmiklos** [comment]
Don't you have some hints about the mentioned website? I looked around the meerun.com, but I have not found and technical info. By looking at the QML files it seems that he used Kalman filtering, and some custom removal of the unlikely movements.

**Comment by jdrescher2006** [comment]
I saw this here: [here](https://talk.maemo.org/showthread.php?t=89556). Have you asked hin directly? Maybe he would provide some infos.

**Comment by martonmiklos** [comment]
Yes I have peeked through this thread (it was fun to read the then existing issues with Nokia's QA acceptance.) I did not found any relevant info other that he used Kalman filter and other methods, but that turns out from the QML codes as well. So I sent him a nice mail and hope the best.

**Comment by jdrescher2006** [comment]
In the first post he lists 3 things:
- Kalman filter
- Remove irrelevant moves (small moves below the accuracy level)
- Removal of unlikely positions

The removing of irrelevant moves should be easy. The distance is calculated between 2 GPS data points. If this distance is below accurate level, ingnore the distance.

Removing unlikely positions is a bit more difficult. For example you could say that biking or running never reaches speeds of 1000km/h or above.
